clouditor-engine-core/src/test/java/io/clouditor/rest/DiscoveryResourceTest.java

```diff
@@ -0,0 +1,195 @@
+package io.clouditor.rest;
+
+import io.clouditor.Engine;
+import io.clouditor.discovery.DiscoveryService;
+import io.clouditor.discovery.Scan;
+import java.util.List;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.Application;
+import javax.ws.rs.core.Response;
+import org.glassfish.jersey.test.JerseyTest;
+import org.glassfish.jersey.test.TestProperties;
+import org.junit.jupiter.api.*;
+
+class DiscoveryResourceTest extends JerseyTest {
+  private static final Engine engine = new Engine();
+  private String token;
+  private static final String targetPrefix = "/discovery/";
+  private static final String FAKE_ID = "fake";
+  private static DiscoveryService discoveryService;
+
+  /* Test Settings */
+  @BeforeAll
+  static void startUpOnce() {
+    engine.setDbInMemory(true);
+
+    engine.setDBName("DiscoveryResourceTestDB");
+
+    // init db
+    engine.initDB();
+
+    // initialize every else
+    engine.init();
+
+    // start the DiscoveryService
+    engine.getService(DiscoveryService.class).start();
+
+    // Initialize discoveryService
+    discoveryService = engine.getService(DiscoveryService.class);
+  }
+
+  @AfterAll
+  static void cleanUpOnce() {
+    engine.shutdown();
+  }
+
+  @BeforeEach
+  public void setUp() throws Exception {
+    super.setUp();
+
+    client().register(ObjectMapperResolver.class);
+
+    if (this.token == null) {
+      this.token = engine.authenticateAPI(target(), "clouditor", "clouditor");
+    }
+  }
+
+  @Override
+  protected Application configure() {
+    // Find first available port.
+    forceSet(TestProperties.CONTAINER_PORT, "0");
+    return new EngineAPI(engine);
+  }
+
+  /* Tests */
+  @Test
+  void testGetScans_whenOneScannerAvailable_thenStatusOkAndResponseNotEmpty() {
+    // Request
+    Response response =
+        target(targetPrefix)
+            .request()
+            .header(
+                AuthenticationFilter.HEADER_AUTHORIZATION,
+                AuthenticationFilter.createAuthorization(token))
+            .get();
+
+    // Assertions
+    List<Scan> scans = response.readEntity(List.class);
+    Assertions.assertFalse(scans.isEmpty());
+    Assertions.assertEquals(Response.Status.OK.getStatusCode(), response.getStatus());
+  }
+
+  @Test
+  void testGetScan_whenRequestedScannerAvailable_thenStatusOkAndRespondIt() {
+    // Request
+    Response response =
```

Comment:
Just FYI, we are using a fairly recent version of Java, so you could also just use `var` instead of explicit type declarations, but its of course not forbidden.

Reply:
Also done. At least for the Response declarations.